### PR TITLE
feat(calc): formulas fallback + 18 new builtins (lookup & conditional)

### PR DESCRIPTION
**Pull request by wolfiesch**
### Commit 1: formulas library fallback + 10 builtins
- Wire `formulas` library as fallback for functions not in builtins (PMT, SLN, NPV, VLOOKUP, etc.)
- 10 new builtins: ROUNDDOWN, MOD, POWER, SQRT, SIGN, LEFT, RIGHT, MID, LEN, CONCATENATE
- `range_shape()` utility for 2D range metadata

### Commit 2: 8 lookup & conditional aggregation builtins
- **8 new builtins**: INDEX, MATCH, XLOOKUP, CHOOSE, SUMIF, SUMIFS, COUNTIF, COUNTIFS
- **RangeValue dataclass**: shape-aware 2D range container preserving `(n_rows, n_cols)` while staying iterable for backward compat
- **`&` string concatenation operator**: enables `=SUMIF(A1:A5, ">"&B1, C1:C5)` dynamic criteria
- **Criteria matching engine**: shared parser for operator prefixes, wildcards, and exact match
- **Whitelist**: 39 → 43 | **Builtins**: 25 → 33

## Why

These functions are required for LRBench-Agent scoring where agents build spreadsheets with formulas. The nested `INDEX(range, MATCH(...))` pattern is ubiquitous in financial models and needs both functions as builtins.

## Test plan

- [x] 298 total tests pass (255 existing + 43 new)
- [x] New tests cover all 8 builtins, `&` operator, criteria engine, RangeValue backward compat, and perturbation propagation
- [x] SynthGL perturbation verifier (12 tests) still passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: python/wolfxl/calc/_functions.py [hunk 1]
```diff
@@ -218,6 +218,118 @@ def _builtin_average(args: list[Any]) -> float:
...
+    return nums[0] - nums[1] * math.floor(nums[0] / nums[1])
+
+
+def _builtin_power(args: list[Any]) -> float:
+    if len(args) != 2:
+        raise ValueError("POWER requires exactly 2 arguments")
+    nums = _coerce_numeric(args)
+    if len(nums) != 2:
+        raise ValueError("POWER: non-numeric argument")
+    return nums[0] ** nums[1]
```
[the remark this answers is not in the document]
Comment (wolfiesch) [reply python/wolfxl/calc/_functions.py]: Fixed in 73831a1. POWER now returns `#NUM!` for negative base with fractional exponent. Test added.

File: python/wolfxl/calc/_functions.py [hunk 2]
```diff
@@ -218,6 +218,118 @@ def _builtin_average(args: list[Any]) -> float:
...
+    return text[-num_chars:] if num_chars > 0 else ""
+
+
+def _builtin_mid(args: list[Any]) -> str:
+    if len(args) != 3:
+        raise ValueError("MID requires exactly 3 arguments")
+    text = _coerce_string(args[0])
+    start = int(_coerce_numeric([args[1]])[0])
+    num_chars = int(_coerce_numeric([args[2]])[0])
+    # Excel MID is 1-indexed
```
[the remark this answers is not in the document]
Comment (wolfiesch) [reply python/wolfxl/calc/_functions.py]: Fixed in 73831a1. MID now returns `#VALUE\!` for start < 1 or num_chars < 0. Two edge case tests added.

File: python/wolfxl/calc/_functions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wolfiesch) [reply python/wolfxl/calc/_functions.py]: Fixed in 73831a1. POWER now guards against complex results and returns `#NUM\!`.

File: python/wolfxl/calc/_evaluator.py [hunk 3]
```diff
@@ -545,6 +580,112 @@ def _resolve_arg(self, arg: str, sheet: str) -> Any:
...
+    # ------------------------------------------------------------------
+
+    def _formulas_fallback(self, formula: str, sheet: str) -> Any:
+        """Evaluate a formula via the ``formulas`` library.
+
+        Compiles the formula into a callable, resolves its cell reference
+        parameters from ``_cell_values``, and returns the scalar result.
+        """
+        import formulas as fm
+        import numpy as np
```
[the remark this answers is not in the document]
Comment (wolfiesch) [reply python/wolfxl/calc/_evaluator.py]: The lazy import is intentional - `formulas` and `numpy` are optional extras (`wolfxl[calc]`). This method only runs as a fallback for functions without builtins. Module-level caching would add complexity for marginal benefit since Python already caches imported modules in `sys.modules`.

File: python/wolfxl/calc/_functions.py [hunk 4]
```diff
@@ -218,6 +218,118 @@ def _builtin_average(args: list[Any]) -> float:
...
+    if val is None:
+        return ""
+    return str(val)
+
+
+def _builtin_left(args: list[Any]) -> str:
+    if len(args) < 1 or len(args) > 2:
+        raise ValueError("LEFT requires 1 or 2 arguments")
+    text = _coerce_string(args[0])
+    num_chars = int(_coerce_numeric([args[1]])[0]) if len(args) > 1 else 1
```
[the remark this answers is not in the document]
Comment (wolfiesch) [reply python/wolfxl/calc/_functions.py]: Fixed in 73831a1. LEFT now returns `#VALUE\!` for negative num_chars, with test.
